### PR TITLE
GCE only system test

### PR DESCRIPTION
**Pull request by dhermes**
~~NOTE: Has #415 as a diffbase.~~

### CODE REVIEW

File: scripts/run_gce_system_tests.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import unittest
+
+import httplib2
+from six.moves import http_client
+
+from oauth2client.client import GoogleCredentials
+from oauth2client.contrib.gce import AppAssertionCredentials
+
+
+SCOPE = ('https://www.googleapis.com/auth/plus.login',
```
Comment (theacodes) [inline scripts/run_gce_system_tests.py]: what's this used for?

Comment (dhermes) [inline scripts/run_gce_system_tests.py]: Copy-pasta. Nice find. (I want #212 to be done so we can turn on `pylint` and 100% coverage so the machines can yell at the humans.)

File: scripts/run_gce_system_tests.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+USER_INFO = 'https://www.googleapis.com/oauth2/v2/userinfo'
+
+
+class TestComputeEngine(unittest.TestCase):
+
+    def test_application_default(self):
+        default_creds = GoogleCredentials.get_application_default()
+        self.assertIsInstance(default_creds, AppAssertionCredentials)
+
+    def test_user_info(self):
```
Comment (theacodes) [inline scripts/run_gce_system_tests.py]: userinfo requires special scopes, but you could use tokeninfo instead.

Comment (dhermes) [inline scripts/run_gce_system_tests.py]: :+1: I was just thinking about this when I deleted the scopes.

Comment (theacodes) [inline scripts/run_gce_system_tests.py]: Example here: https://github.com/google/oauth2client/blob/master/oauth2client/client.py#L988

Comment (dhermes) [inline scripts/run_gce_system_tests.py]: @jonparrott I have to be AFK for a bit. You want to just fork this PR and run with it?

Comment (theacodes) [inline scripts/run_gce_system_tests.py]: How long is a bit? I likely won't be able to do that today, but happy to poke at it tomorrow.

Comment (dhermes) [inline scripts/run_gce_system_tests.py]: 3+ hours. I can jump on it once I get home tonight.

**Comment by dhermes** [comment]
@jonparrott PTAL

**Comment by jonparrott** [comment]
this LGTM, @nathanielmanistaatgoogle?

**Comment by nathanielmanistaatgoogle** [comment]
Why isn't Travis showing the test having been run? Is a `travis.yml` update due in this change?

**Comment by dhermes** [comment]
Travis doesn't run in GCE. This Travis environment will fail anywhere but on GCE.

**Comment by nathanielmanistaatgoogle** [comment]
Oh right...

**Comment by dhermes** [comment]
See my and @jonparrott [previous discussion][1]. (And for future people who make it here.)

[1]: https://github.com/google/oauth2client/pull/387#issuecomment-185349768

**Comment by dhermes** [comment]
OK @jonparrott LMK if I can do anything to help getting the Jenkins test going.

**Comment by jonparrott** [comment]
Will do. Setting it up now.

**Comment by dhermes** [comment]
w00t!

**Pull request merged**